lib/DataObject/GridColumnConfig/Operator/Factory/TranslateValueFactory.php

```diff
@@ -35,6 +36,9 @@ public function __construct(TranslatorInterface $translator)
 
     public function build(\stdClass $configElement, $context = null): OperatorInterface
     {
+	    if(null !== $context && isset($context['language']) && Tool::isValidLanguage($context['language'])) {
+		    $this->translator->setLocale($context['language']);
```

Comment:
I would pass both translator service and locale to the TranslateValue operator and set + restore the locale in getLabeledValue similar to https://github.com/pimcore/pimcore/blob/c62924ca5c3d0c482a71097cb0231aff143208d6/lib/DataObject/GridColumnConfig/Operator/LocaleSwitcher.php#L54

Reply:
We are not sure exactly what you mean. The LocalSwitcherOperator gets the language from the operator setting while the TranslatedValueOperator gets the language from the grid configuration / request. The Pimcore\Model\DataObject\Service receives the language from the request. The only possibility we saw was to pass this parameter in context.
Would the solution shown on the screenshot be OK?
<img width="1591" alt="Bildschirmfoto 2019-05-21 um 15 39 50" src="https://user-images.githubusercontent.com/16014748/58101365-c5ebd000-7bdf-11e9-8fb7-26f438f7447e.png">

Reply:
yes :-)